### PR TITLE
Silabs gpio

### PR DESCRIPTION
**Pull request by galak**
Convert silabs gpio driver over to dts, and the boards as well.

### CODE REVIEW

**Review by chrta — COMMENTED** [review]
How can i test this, especially th button flags? The basic button example and for LED the basic blinky example?

**Comment by galak** [comment]
> How can i test this, especially th button flags? The basic button example and for LED the basic blinky example?

if you build the two samples and see if the LED or button work.  Or are you asking something else?

**Comment by galak** [comment]
> Result: Both examples work fine on both boards.

Nope, that's the best & easiest way to test.

File: soc/arm/silabs_exx32/efm32wg/dts.fixup [hunk 1]
```diff
@@ -12,4 +12,17 @@
 #define CONFIG_UART_GECKO_0_BAUD_RATE	SILABS_EFM32_USART_4000C000_CURRENT_SPEED
 #define CONFIG_UART_GECKO_0_IRQ_PRI	SILABS_EFM32_USART_4000C000_IRQ_0_PRIORITY
 
+#define CONFIG_GPIO_GECKO_COMMON_NAME	SILABS_EFM32_GPIO_40006100_LABEL
+#define CONFIG_GPIO_GECKO_COMMON_EVEN_IRQ	SILABS_EFM32_GPIO_40006100_IRQ_GPIO_EVEN
+#define CONFIG_GPIO_GECKO_COMMON_EVEN_PRI	SILABS_EFM32_GPIO_40006100_IRQ_GPIO_EVEN_PRIORITY
+#define CONFIG_GPIO_GECKO_COMMON_ODDN_IRQ	SILABS_EFM32_GPIO_40006100_IRQ_GPIO_ODD
```
[the remark this answers is not in the document]
Comment (galak) [reply soc/arm/silabs_exx32/efm32wg/dts.fixup]: Fixed

File: dts/arm/silabs/efm32wg.dtsi [hunk 2]
```diff
@@ -61,6 +62,66 @@
 			status = "disabled";
 			label = "UART_4";
 		};
+
+		gpio@40006100 {
+			compatible = "silabs,efm32-gpio";
+			reg = <0x40006100 0xf00>;
+			interrupts = <1 0 11 0>;
```
[the remark this answers is not in the document]
Comment (galak) [reply dts/arm/silabs/efm32wg.dtsi]: Fixed

**Review by chrta — APPROVED** [review]

**Pull request merged**